deploy/deploy.go

```diff
@@ -594,3 +603,96 @@ func (m *MeshnetSpec) Healthy(ctx context.Context) error {
 		}
 	}
 }
+
+type IxiaTGSpec struct {
+	ManifestDir string           `yaml:"manifests"`
+	ConfigMap   *IxiaTGConfigMap `yaml:"configMap"`
+	kClient     kubernetes.Interface
+}
+
+type IxiaTGConfigMap struct {
+	Release string         `yaml:"release" json:"release"`
+	Images  []*IxiaTGImage `yaml:"images" json:"images"`
+}
+
+type IxiaTGImage struct {
+	Name string `yaml:"name" json:"name"`
+	Path string `yaml:"path" json:"path"`
+	Tag  string `yaml:"tag" json:"tag"`
+}
+
+func (i *IxiaTGSpec) SetKClient(c kubernetes.Interface) {
+	i.kClient = c
+}
+
+func (i *IxiaTGSpec) Deploy(ctx context.Context) error {
+	log.Infof("Deploying IxiaTG controller from: %s", i.ManifestDir)
+	if err := execer.Exec("kubectl", "apply", "-f", filepath.Join(i.ManifestDir, "ixiatg-operator.yaml")); err != nil {
+		return err
+	}
+	if i.ConfigMap == nil {
+		path := filepath.Join(i.ManifestDir, "ixia-configmap.yaml")
+		if _, err := os.Stat(path); err == nil {
+			log.Infof("Deploying IxiaTG configmap from: %s", path)
+			if err := execer.Exec("kubectl", "apply", "-f", path); err != nil {
+				return err
+			}
+		}
+		log.Infof("IxiaTG controller Deployed")
+		return nil
+	}
+	b, err := json.MarshalIndent(i.ConfigMap, "    ", "  ")
+	if err != nil {
+		return err
+	}
+	b = append([]byte(ixiaTGConfigMapHeader), b...)
+	f, err := os.CreateTemp("", "ixiatg-configmap-*.yaml")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(f.Name())
+	if _, err := f.Write(b); err != nil {
+		return err
+	}
+	log.Infof("Deploying IxiaTG configmap from: %s", f.Name())
+	if err := execer.Exec("kubectl", "apply", "-f", f.Name()); err != nil {
+		return err
+	}
+	log.Infof("IxiaTG controller Deployed")
+	return nil
+}
+
+func (i *IxiaTGSpec) Healthy(ctx context.Context) error {
+	return deploymentHealthy(ctx, i.kClient, "ixiatg-op-system")
+}
+
+func deploymentHealthy(ctx context.Context, c kubernetes.Interface, name string) error {
+	log.Infof("Waiting on deployment %q to be healthy", name)
+	w, err := c.AppsV1().Deployments(name).Watch(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	ch := w.ResultChan()
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("context canceled before healthy")
+		case e, ok := <-ch:
+			if !ok {
+				return fmt.Errorf("watch channel closed before healthy")
+			}
+			d, ok := e.Object.(*appsv1.Deployment)
+			if !ok {
+				return fmt.Errorf("invalid object type: %T", d)
+			}
+			if d.Status.AvailableReplicas == 1 &&
```

Comment:
Is 1 & 0 here true & false, or it indicates the number of replica in terms of state? If it is the latter, I think it might have dependencies on the config?